src/integrationTest/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/OrganisationServiceIntegrationTest.java

```diff
@@ -0,0 +1,138 @@
+package uk.gov.hmcts.reform.orgrolemapping.domain.service;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.hmcts.reform.orgrolemapping.controller.BaseTestIntegration;
+import uk.gov.hmcts.reform.orgrolemapping.data.OrganisationRefreshQueueRepository;
+import uk.gov.hmcts.reform.orgrolemapping.data.ProfileRefreshQueueRepository;
+import uk.gov.hmcts.reform.orgrolemapping.domain.model.OrganisationInfo;
+import uk.gov.hmcts.reform.orgrolemapping.domain.model.OrganisationByProfileIdsResponse;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+@Transactional
+public class OrganisationServiceIntegrationTest extends BaseTestIntegration {
+
+    @Autowired
+    private OrganisationService organisationService;
+
+    @Autowired
+    private ProfileRefreshQueueRepository profileRefreshQueueRepository;
+
+    @Autowired
+    private OrganisationRefreshQueueRepository organisationRefreshQueueRepository;
+
+    @MockBean
+    private PrdService prdService;
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+            scripts = {"classpath:sql/insert_profile_refresh_queue.sql"})
+    void shouldInsertOneOrganisationIntoOrganisationRefreshQueue_AndClearProfileRefreshQueue() {
+        OrganisationInfo organisationInfo = OrganisationInfo.builder()
+                .organisationIdentifier("123")
+                .status("ACTIVE")
+                .lastUpdated(LocalDateTime.now())
+                .organisationProfileIds(List.of("SOLICITOR_PROFILE")).build();
+
+        OrganisationByProfileIdsResponse response = OrganisationByProfileIdsResponse.builder()
+                .organisationInfo(List.of(organisationInfo))
+                .lastRecordInPage("123")
+                .moreAvailable(false).build();
+
+        when(prdService.fetchOrganisationsByProfileIds(any(), eq(null), any()))
+                .thenReturn(ResponseEntity.ok(response));
+
+        organisationService.findAndInsertStaleOrganisationsIntoRefreshQueue();
+
+        assertEquals(organisationRefreshQueueRepository.findAll().size(), 1);
+        assertEquals(profileRefreshQueueRepository.getActiveProfileEntities().size(), 0);
+    }
+
+    @Test
+    @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD,
+            scripts = {"classpath:sql/insert_profile_refresh_queue.sql"})
+    void shouldInsertTwoOrganisationIntoOrganisationRefreshQueue_AndClearProfileRefreshQueue() {
```

Comment:
I wonder if this test is overkill? Is there any DB integrity you are checking by having 2 orgs as opposed to 1 org?

Reply:
have removed IT in commit [d877fc1](https://github.com/hmcts/am-org-role-mapping-service/pull/1743/commits/d877fc137bdf20ec1516907aacb376c4e7142fca)